File: 26.MemoryLeak/app/src/main/java/com/application/memoryleak/MainActivity.java
package com.application.memoryleak;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import java.lang.ref.WeakReference;

public class MainActivity extends AppCompatActivity {

    SomeRandomSampleClass someRandomSampleClass;
    private static WeakReference<MainActivity> activityWeakReference;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        someRandomSampleClass = SomeRandomSampleClass.getSomeRandomSampleClass(this);
    }
}

File: 20.AndroidLoginSignup_With_MVVM/app/src/main/java/com/example/loginsignup/MainActivity.kt
package com.example.loginsignup

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.loginsignup.ui.auth.AuthActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        finish()
        startActivity(Intent(this,AuthActivity::class.java))
    }
}